code/modules/projectiles/modular/laser_base.dm

```diff
@@ -0,0 +1,215 @@
+/obj/item/laser_components
+	icon = 'icons/obj/modular_laser.dmi'
+	icon_state = "bfg"
+	var/reliability = 0
+	var/damage = 1
+	var/fire_delay = 0
+	var/condition = 0 //inverse health of the component. subtracted from reliability.
+	var/shots = 0
+	var/burst = 0
+	var/accuracy = 0
+	var/obj/item/weapon/repair_item
+	var/gun_overlay
+
+/obj/item/laser_components/proc/degrade(var/increment = 1)
+	if(increment)
+		condition += increment
+		if(condition > reliability)
+			condition = reliability
+
+/obj/item/laser_components/attackby(var/obj/item/weapon/D as obj, var/mob/user as mob)
+	if(!istype(D,repair_item))
+		return ..()
+	user << "<span class='warning'>You begin repairing \the [src].</span>"
+	if(do_after(user,20) && negadegrade(D))
+		user << "<span class='notice'>You repair \the [src].</span>"
+	else
+		user << "<span class='warning'>You fail to repair \the [src].</span>"
+
+/obj/item/laser_components/proc/negadegrade(var/obj/item/weapon/D)
+	return 1
+
+/obj/item/laser_components/modifier
+	name = "modifier"
+	desc = "A basic laser weapon modifier."
+	reliability = -5
+	var/mod_type
+	var/base_malus = 2 //when modifiers get damaged they do not break, but make other components break faster
+	var/malus = 2 //subtracted from weapon's overall reliability everytime it's fired
+	var/gun_force = 0 //melee damage of the gun
+	var/chargetime = 0
+	var/burst_delay = 0
+	var/scope_name
+	var/criticality
+	repair_item = /obj/item/weapon/weldingtool
+
+/obj/item/laser_components/modifier/examine(mob/user)
+	. = ..(user, 1)
+	if(.)
+		if(malus > base_malus)
+			user << "<span class='warning'>\The [src] appears damaged.</span>"
+
+/obj/item/laser_components/modifier/degrade(var/increment = 1)
+	if(increment)
+		malus += increment
+		if(malus > abs(base_malus*2))
+			malus = abs(base_malus*2)
+
+/obj/item/laser_components/modifier/negadegrade(var/obj/item/weapon/D)
+	var/obj/item/weapon/weldingtool/W = D
+	if(W.remove_fuel(5))
+		malus -= 5
+		if(malus < base_malus)
+			malus = base_malus
+		return 1
+	return 0
+
+/obj/item/laser_components/capacitor
+	name = "capacitor"
+	desc = "A basic laser weapon capacitor."
+	icon_state = "capacitor"
+	shots = 5
+	damage = 10
+	reliability = 50
+	repair_item = /obj/item/stack/cable_coil
+
+/obj/item/laser_components/capacitor/negadegrade(var/obj/item/weapon/D)
+	var/obj/item/stack/cable_coil/C = D
+	if(C.amount)
+		C.amount -= 5
```

Comment:
There has to be a proc for this. Right now this can result in C.amount being set to a negative number.